feature/metis/conversation/src/test/kotlin/de/tum/informatics/www1/artemis/native_app/feature/metis/conversation/ConversationAnswerMessagesUITest.kt

```diff
@@ -0,0 +1,202 @@
+
+        assert(resolvedPost != null)
+        assert(resolvedPost is AnswerPostPojo)
+        assert((resolvedPost as AnswerPostPojo).content == answers[2].content!!)
+    }
+
+    @Test
+    fun `test GIVEN post is resolved WHEN un-resolving the post THEN the post is un-resolved`() {
+        val resolvingIndex = 0
+
+        val modifiedAnswers = answers.toMutableList()
+        modifiedAnswers[resolvingIndex] = modifiedAnswers[resolvingIndex].copy(resolvesPost = true)
+        val resolvedPost = post.copy(
+            resolved = true,
+            answers = modifiedAnswers
+        )
+
+        var unresolvedPost: IBasePost? = null
+
+        setupUi(resolvedPost) { post ->
+            unresolvedPost = post
+            CompletableDeferred()
+        }
+
+        composeTestRule.onNodeWithText(answers[resolvingIndex].content!!, useUnmergedTree = true)
+            .performSemanticsAction(SemanticsActions.OnLongClick)
+        composeTestRule.onNodeWithText(context.getString(R.string.post_does_not_resolve))
+            .performClick()
+
+        assert(unresolvedPost != null)
+        assert(unresolvedPost is AnswerPostPojo)
+        assert((unresolvedPost as AnswerPostPojo).content == answers[resolvingIndex].content!!)
+    }
+
+    @Test
+    fun `test GIVEN the post is not resolved and no answer post is resolving THEN the post is shown as not resolved and no answer post is shown as resolving`() {
+        setupUi(post) { CompletableDeferred() }
+
+        composeTestRule.onNodeWithText(post.content).assertExists()
+        for (answer in answers) {
+            composeTestRule.onNodeWithText(answer.content!!).assertExists()
+        }
+        composeTestRule.onNodeWithText(context.getString(R.string.post_is_resolved))
+            .assertDoesNotExist()
+        composeTestRule.onNodeWithText(context.getString(R.string.post_resolves))
+            .assertDoesNotExist()
+    }
+
+    @Test
+    fun `test GIVEN the post is resolved and one answer post is marked as resolving THEN the post is shown as resolved and this answer post is shown as resolving`() {
+        val resolvingIndex = 0
+
+        val modifiedAnswers = answers.toMutableList()
+        modifiedAnswers[resolvingIndex] = modifiedAnswers[resolvingIndex].copy(resolvesPost = true)
+        val resolvedPost = post.copy(
+            resolved = true,
+            answers = modifiedAnswers
+        )
+
+        setupUi(resolvedPost) { CompletableDeferred() }
+
+        composeTestRule.onNodeWithText(post.content).assertExists()
+        composeTestRule.onNodeWithText(context.getString(R.string.post_is_resolved)).assertExists()
+        composeTestRule.onNodeWithText(context.getString(R.string.post_resolves)).assertExists()
+        composeTestRule
+            .onNodeWithText(answers[resolvingIndex].content!!, useUnmergedTree = true)
+            .onParent()
+            .assert(hasAnyChild(hasText(context.getString(R.string.post_resolves))))
```

Comment:
This code seems rather fragile. I guess onParent could break quite easily, should the layout of the answer post be changed or refactored in the future. I would set a testTag on all answer post composables and look for that instead. 

Also, this code only checks that the post_resolves is displayed on the resolvingIndex post, but it lacks a check that it is *not* displayed on all the other answer posts.